### PR TITLE
Session cache fix

### PR DESCRIPTION
**Pull request by pcantrell**
Alex noted a bug where sessions would show as favorited (Minnestar icon next to them) even though the user hadn't favorited them. I believe this fixes the issue.

I also threw in a fix for a server error when logging out redundantly (log out, back, log out again).

### CODE REVIEW

File: src/app/views/pages/_sessions.html.erb [hunk 1]
```diff
@@ -1,4 +1,4 @@
-  <%= cache Session.maximum(:updated_at), expires_in: 20.seconds do %>
+  <%= cache [Session.maximum(:updated_at), current_participant], expires_in: 20.seconds do %>
```
Comment (jcoyne) [inline src/app/views/pages/_sessions.html.erb]: Would you add a comment here about why we're using current participant as part of the cache key

File: src/app/views/sessions/index.html.erb [hunk 2]
```diff
@@ -1,7 +1,7 @@
 <% title('All Sessions') %>
 (most recently added first)
 
-<% cache @sessions do %>
+<% cache [@sessions, current_participant] do %>
```
Comment (jcoyne) [inline src/app/views/sessions/index.html.erb]: Also add a comment here.

Comment (pcantrell) [inline src/app/views/sessions/index.html.erb]: The comment’s just going to say “fragment contains user-specific info,” which I feel like the cache key itself already conveys. But I’ll add it.

**Comment by pcantrell** [comment]
Thanks for the speedy turnaround!

File: src/app/views/pages/_sessions.html.erb [hunk 3]
```diff
@@ -1,4 +1,5 @@
-  <%= cache Session.maximum(:updated_at), expires_in: 20.seconds do %>
+  <%# Note that fragment contains user-specific info, must be cached per user %>
+  <%= cache [Session.maximum(:updated_at), current_participant], expires_in: 20.seconds do %>
```
Comment (jcoyne) [inline src/app/views/pages/_sessions.html.erb]: Where is the user-specific info? I can't find it. What partial?

Comment (pcantrell) [inline src/app/views/pages/_sessions.html.erb]: Here: https://github.com/minnestar/sessionizer/blob/master/src/app/views/sessions/_session.html.erb#L1

`attending_class(session)` gives results based on the current user.

**Pull request merged**